mesonbuild/mesonlib.py

```diff
@@ -346,6 +346,11 @@ def __setitem__(self, machine: MachineChoice, val: typing.Optional[_T]) -> None:
         }[machine]
         setattr(self, key, val)
 
+# BBI_SOL11_4 BEGIN -- added definition for platform specific conditions
+def is_sunos() -> bool:
+    return platform.system().lower() == 'sunos'
```

Comment:
there was some debate a while ago whether we should call this `sunos` or `solaris`, @jpakkane, I'm okay with sunos at this point and I'm willing to just make the call, does that work for you?